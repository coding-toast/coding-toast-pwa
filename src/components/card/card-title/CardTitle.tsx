import * as React from 'react'
import CardTitleStyles from './CardTitle.module.scss'

interface CardTitleProps {
  title: string
  backgroundUrl?: string
}

const CardTitle: React.FC<CardTitleProps> = props => {
  const { title, backgroundUrl } = props

  const backgroundImage = {
    backgroundImage: `url(${backgroundUrl})`,
    backgroundSize: 'cover',
    marginBottom: '1em'
  }

  const cardTitleHeightWithBackground = {
    height: '125px'
  }

  return (
    <div className={CardTitleStyles.cardTitleContainer}>
      {backgroundUrl ? <div style={backgroundImage} className={CardTitleStyles.cardTitleBackground} /> : ''}
      <h2 style={backgroundUrl ? cardTitleHeightWithBackground : {}} className={`${CardTitleStyles.cardTitle} text-color-primary`}>
        {title}
      </h2>
    </div>
  )
}

export default CardTitle
