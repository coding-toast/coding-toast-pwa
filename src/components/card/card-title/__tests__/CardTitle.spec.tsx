import React from 'react'
import renderer from 'react-test-renderer'

import CardTitle from '../CardTitle'

describe('<CardTitle />', () => {
  it('renders correctly without a background', () => {
    const tree = renderer.create(<CardTitle title="Lorem Ipsum" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with a background', () => {
    const tree = renderer.create(<CardTitle title="Lorem Ipsum" backgroundUrl="test url" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
