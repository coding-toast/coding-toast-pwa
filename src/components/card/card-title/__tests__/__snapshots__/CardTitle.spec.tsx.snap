// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTitle /> renders correctly with a background 1`] = `
<div
  className="cardTitleContainer"
>
  <div
    className="cardTitleBackground"
    style={
      Object {
        "backgroundImage": "url(test url)",
        "backgroundSize": "cover",
        "marginBottom": "1em",
      }
    }
  />
  <h2
    className="cardTitle text-color-primary"
    style={
      Object {
        "height": "125px",
      }
    }
  >
    Lorem Ipsum
  </h2>
</div>
`;

exports[`<CardTitle /> renders correctly without a background 1`] = `
<div
  className="cardTitleContainer"
>
  <h2
    className="cardTitle text-color-primary"
    style={Object {}}
  >
    Lorem Ipsum
  </h2>
</div>
`;
